hooks/tk-multi-publish2/maya/publish_session_model.py

```diff
@@ -99,3 +131,46 @@ def validate(self, task_settings, item):
                         self.logger.warning("Failed modelpublish validation: {}".format(check))
 
         return super(MayaPublishSessionModelPlugin, self).validate(task_settings, item)
+
+    def publish_files(self, task_settings, item, publish_path):
+        """
+        This method publishes (copies) the item's path property to the publish location.
+        For session override this to do cleanup and save to publish location instead,
+        then discard the changes done during cleanup from the workfile so that they are
+        not preserved while versioning up.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :param publish_path: The output path to publish files to
+        """
+
+        path = copy.deepcopy(item.properties.get("path"))
+        if not path:
+            raise KeyError("Base class implementation of publish_files() method requires a 'path' property.")
+
+        # Save to publish path
+        self.cleanup_file(item)
+        self._save_session(publish_path, item)
```

Comment:
FYI, the new version of `self._save_session()` also requires a `version` argument.

Reply:
sgtk_apps-3.18.0_bug1258